nixos/modules/profiles/macos-builder.nix

```diff
@@ -67,9 +67,9 @@ in
        '';
     };
     hostPort = mkOption {
-      default = 22;
+      default = 31022;
```

Comment:
I also hope it's possible to separate properties like host port, memory, and others from the cached builder. I'm invested in this too so that I can use my Mac to test or review Linux packages and can spend some time on it too, but only if you desire it.

Reply:
Yeah, changing the `hostPort` won't invalidate the cached VM build

I also feel like we should keep the default value to 22 if we can to avoid breaking existing configurations

Reply:
I've updated this PR and added a way for consumers to be able to override `darwin.builder` more easily:

[darwin.builder: allow overriding configuration](https://github.com/NixOS/nixpkgs/commit/dac055259ad4b2972abbffd805b672e40d912c2a)

My preference would be changing the default so that future users don't have to deal with exposing the VM on port 22, and with the above change, existing users can choose to use the existing port or switch to a different port with the updated instructions in the documentation.

Reply:
Alternatively, we could not set a default port, and let the users configure this using an `override` or through the module system.

Although it would break the following command:

```
$ nix run nixpkgs#darwin.builder
```

Reply:
We could rename it to `darwin.linuxBuilder` and inform users about the change using the deprecation message.

Not a repeatable solution though... Does anyone have other breaking changes in mind? Should we wait?

Reply:
Oh actually we might change it again to use package name syntax instead of general identifier syntax: `linux-builder`. Awkward but useful? Maybe we should just give it a version at some point.

Reply:
I'm happy to rename this and add the deprecation message, is there a preference between `linuxBuilder` or `linux-builder`?

Reply:
Let's go with `linux-builder`.